src/uwtools/drivers/driver.py

```diff
@@ -410,12 +487,15 @@ def _scheduler(self) -> JobScheduler:
         """
         return JobScheduler.get_scheduler(self._resources)
 
-    def _validate(self) -> None:
+    def _validate(self, schema_file: Optional[Path] = None) -> None:
         """
         Perform all necessary schema validation.
         """
-        for schema_name in (self._driver_name.replace("_", "-"), "platform"):
-            validate_internal(schema_name=schema_name, config=self._config)
+        if schema_file:
+            validate_external(schema_file=schema_file, config=self._config)
+        else:
+            for schema_name in (self._driver_name.replace("_", "-"), "platform"):
+                validate_internal(schema_name=schema_name, config=self._config)
```

Comment:
```suggestion
        if schema_file:
            validate_external(schema_file=schema_file, config=self._config)
        else:
            validate_internal(schema_name=self._driver_name.replace("_", "-"), config=self._config)
        validate_internal(schema_name="platform", config=self._config)
```
I think we always want to validate the `platform:` block against its `uwtools`-internal schema.